quickstarts/microsoft.compute/vm-simple-windows/main.bicep

```diff
@@ -240,10 +231,31 @@ resource vm 'Microsoft.Compute/virtualMachines@2021-03-01' = {
     diagnosticsProfile: {
       bootDiagnostics: {
         enabled: true
-        storageUri: stg.properties.primaryEndpoints.blob
+        storageUri: storageAccount.properties.primaryEndpoints.blob
+      }
+    }
+    securityProfile: ((securityType == 'TrustedLaunch') ? securityProfileJson : json('null'))
+  }
+}
+
+resource vmExtension 'Microsoft.Compute/virtualMachines/extensions@2022-03-01' = if (secureBoot) {
+  parent: vm
+  name: extensionName
+  location: location
+  properties: {
+    publisher: extensionPublisher
+    type: extensionName
+    typeHandlerVersion: extensionVersion
+    autoUpgradeMinorVersion: true
+    settings: {
```

Comment:
now that GA extension support autoupgrade, please add autoupgrade property here as well.

"enableAutomaticUpgrade": true,

Reply:
@Bab95 - Deployment pipeline are generating error for Public and Gov Cloud:

Status Message: VM Extension with publisher 'Microsoft.Azure.Security.WindowsAttestation' and type 'GuestAttestation' does not support setting enableAutomaticUpgrade property to true on this subscription. (Code:OperationNotAllowed)